include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -2705,7 +2755,7 @@ class FEFaceEvaluation : public FEEvaluationAccess<dim,
    * static_dofs_per_component, but the number depends on the actual element
    * selected and is thus not static.
    */
-  const unsigned int dofs_per_component;
+  const unsigned int dofs_per_component_on_cell;
```

Comment:
https://github.com/dealii/dealii/blob/49444955f263372e7b44a14f90a4191325749813/include/deal.II/matrix_free/fe_evaluation.h#L7236-L7237

In `FEEvaluation` it is fine, because we only have cell dofs there. Consistent would be `dofs_per_component_on_cell` in both classes (and deprecation of the old name).

Reply:
`dofs_per_component_on_cell` is a rather long name, and for the evaluator there is not really a `dofs_per_[component_on_]face` that is uniquely defined that would motivate using another name. The relevant information is the number of all unknowns on the cell, which is why the particular name was chosen. I see why the change was done but I would be opposed to it because the number of cell dofs and face dofs is not information on the same level of abstraction:

We have the two names you suggest here in `ShapeInfo`; in my opinion, the name `dofs_per_component_on_face` in `ShapeInfo` is poorly chosen and somewhat misleading, so I would rather prefer to also adapt the name in `ShapeInfo`. Overall, the `dofs_per_component` variable is a non-internal variable and potentially often used, so I would only make a deprecation and change if we have a uniformly better naming scheme. I do not see that yet, but I might be missing something here. 
Regarding the name for the faces, I would suggest something along `dofs_per_component_projected_onto_face` or something along those lines that is clear how we use it. I think we need to avoid the `FiniteElement::dofs_per_face` analogy here (which only counts degrees of freedom shared between cells, but not DG ones), because depending on the basis and finite element, the notion of which dofs are active for a particular element degree on a face may vary. In other words, I think we should use a name suggesting the tensor product language of referring to some projection/restriction/interpolation of the cell data onto the face.